### PR TITLE
Computing path matrix from powers of adjacency matrix

### PR DESCRIPTION
**Pull request by VaishnaviMandloi**
## Related Issue
- #1344  

## Proposed Changes
- added implementation code forPath Matrix .
- added a explanation in readme file with image.

## Additional Info
- Computing path matrix from its adjacency matrix by multiplying adjacency matrix with itself , a[i][j] represent the number of paths of path length of p (p = power of matrix) from vertex i yo j.
- changes are made in only G-Graph folder.
-1st file : PathMatrix.cpp
-2nd file: readme.md
-tested with many input , screenshot of one of those is provided below.
-time complexity O(p(n^3)) , where n is total vertex and p is the power of adjacency matrix .

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | 
| ----------- | 
| Output  | 
| 
output #1344
  |

### CODE REVIEW

File: G-Graphs/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        {
+          result[i][j]=adj[i][j]
+        }
+    }
+    for(k : 1 to p ) do
+    {
+        multiply(result,adj,tmp)
+        for (i : 1 to n) do
+        { 
+            for (j : 1 to n) do
```
Comment (Mrjoy832) [inline G-Graphs/README.md:43]: - Add algorithm not pseudo Code
- Add space Complexity
- Add your PR into G-Graph/I-Implementition

File: G-Graphs/PathMatrix.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    for(power=1;power<=n;power++){
+        power_matrix(power,result);
+        cout<<"adjacency matrix raised to power "<<power<<" : "<<endl;
+        display(result);
+        for(i=0;i<n;i++){
+            for(j=0;j<n;j++){ 
+                x[i][j] = x[i][j] + result[i][j]; 
+            }
+        }
+    }
```
Comment (Mrjoy832) [inline G-Graphs/PathMatrix.cpp:35]: Add more comments in Code

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: G-Graphs/PathMatrix.cpp (hunk 2 above)
Comment (VaishnaviMandloi) [reply G-Graphs/PathMatrix.cpp]: kindly review my pr now , changes are made as told 
added comment ,change file location of PathMatrix to Graph-implementation folder and  updated readme of Graph-Implementation folder

**Comment by Mrjoy832** [comment]
3 files are changed why?

**Comment by VaishnaviMandloi** [comment]
> 3 files are changed why?

Readme phele graph folder ki update ki thi ...phir PathMatrix.cpp file Graph-implementation folder mai move kri to readme bhi graph-implementation wale folder ki update kri. Isliye 3 file h 
1. Readme jo graph folder mai h wo empty h
2. Readme jo graph-Implementation folder mai h wo updated h
3. PathMatrix.cpp file

File: G-Graphs/README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<!-- Table of content -->
+# Table of content
```
Comment (Mrjoy832) [inline G-Graphs/README.md:2]: Keep it as same as master branch

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: G-Graphs/README.md (hunk 3 above)
Comment (VaishnaviMandloi) [reply G-Graphs/README.md]: Made the required change , now review  it please..

File: G-Graphs/I-Implementation/Readme.md [hunk 4]
```diff
@@ -13,6 +13,13 @@
     - [Advantage](#advantage)
     - [Disadvantage](#disadvantage)
 
+- [Path Matrix](#path-matrix])
```
Comment (Mrjoy832) [inline G-Graphs/I-Implementation/Readme.md:16]: link not working

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: G-Graphs/I-Implementation/Readme.md (hunk 4 above)
Comment (VaishnaviMandloi) [reply G-Graphs/I-Implementation/Readme.md]: Made the required changes , please check now

**Review by Mrjoy832 — APPROVED** [review]

**Comment by Mrjoy832** [comment]
@Lakhankumawat , as G-Graphs readme is already empty and her changes in Readme of G-Graphs will not affect,So I referred You ..
LGTM🌟

**Review by Lakhankumawat — APPROVED** [review]
LGTM 💻🎉✨

**Comment by Lakhankumawat** [comment]
Mention this pr number in your next pr i will increase prs level , you have continuous good contribution , i value your efforts but i cannot yours prs to level2 continuously so we are having level1 here , hopee you understand ,

**Comment by VaishnaviMandloi** [comment]
> Mention this pr number in your next pr i will increase prs level , you have continuous good contribution , i value your efforts but i cannot give yours prs to level2 continuously so we are having level1 here , hope you understand ,

 

> 

@Lakhankumawat  yea I understand 😁 , exited for more contributions..

**Pull request merged**